include/deal.II/lac/matrix_out.h

```diff
@@ -340,7 +347,21 @@ MatrixOut::build_patches(const Matrix      &matrix,
 
   // first clear old data and re-set the object to a correctly sized state:
   patches.clear();
-  patches.resize((gridpoints_x) * (gridpoints_y));
+  try
+    {
+      patches.resize(gridpoints_x * gridpoints_y);
+    }
+  catch (const std::bad_alloc &)
+    {
+      AssertThrow(false,
+                  ExcMessage("You are trying to create a graphical "
+                             "representation of a matrix that would "
+                             "requiring outputting " +
+                             std::to_string(gridpoints_x) + "x" +
+                             std::to_string(gridpoints_y) +
+                             " patches. There is not enough memory to output " +
+                             "this many patches."));
+    }
```

Comment:
When you try to use this class on a matrix of size `nrows=28672, ncols=28672, nnonzeros=1227952`, we are trying to allocate 822,026,241 patches here. That's too much for my laptop, but all I get is a `std::bad_alloc` exception. We can produce a better error message.